package.nls.json

```diff
@@ -31,6 +31,7 @@
     "configuration.java.test.config.testKind.description": "Specify the targeting test framework for this test configuration. Supported values are `junit`, `testng`.",
     "configuration.java.test.config.filters.description": "Specify the test filters.",
     "configuration.java.test.config.filters.tags.description": "Specify the tags to be included or excluded. \n\nTags having `!` as the prefix will be **excluded**. \n\nNote: This setting **only** takes effect when `testKind` is set to `junit`.",
+    "configuration.java.test.config.when.description": "Specify the [when clause](https://code.visualstudio.com/api/references/when-clause-contexts) for matching tests by to determine if the configuration should be run with.\n\nWhen launching a test that satisfies a single configuration's when clause, it will be run with that configuration. If multiple configurations are satisfied, the user will be prompted to pick which configuration to run with.\n\nWhen launching multiple tests (e.g. for a class or package), a configuration's when clause must be satisfied for **all** tests to be considered.\n\nConfigurations that do not define a when clause will match all tests.",
```

Comment:
I suggest avoiding mentioning the when-clause from VS Code document. Because the context key listed there are not supported here, and we do not support `and`, `or` now. It may confuse users.

Just simply tells user that `testItem ~= /<regular-expression>/` is the only clause we support right now.

If possible, give some examples will be even better, like this setting:
<img width="740" alt="Screenshot 2023-12-15 at 10 49 35" src="https://github.com/microsoft/vscode-java-test/assets/6193897/45bbfa08-7032-453e-afef-cfa8feeccb6a">

Reply:
Do you want examples in the `package.nls.json`, wiki, or both?

Reply:
Yes, just give one or two examples will be good enough.

I'll take care of the package.nls.zh.json & wiki.

Reply:
https://github.com/microsoft/vscode-java-test/pull/1638/commits/9cbdbbdc51ccc8b85bbd5fba272474d44ee34eaf